### DISCUSSION TITLE
How to grab window from 0.19.1 wgpu create_surface?

### DISCUSSION THREAD

**Discussion by lexi-the-cute** [gfx-rs/wgpu#5213] [Q&A] [answered]
I'm trying to make sure I can resize the surface to match my window when the user resizes the window and am not sure how to grab the window's internal size after creating the surface with `instance.create_surface(&window).unwrap();`

**Comment by cwfitzgerald** [comment, marked as answer]
You can't, you need to retain a handle to the window yourself - the general way to do this is to hand us an Arc<Window> and keep an arc for yourself.

**Reply by lexi-the-cute** [reply]
perfect! that was surprisingly easy once i learned about Arc from you. thanks!

**Comment by lexi-the-cute** [comment]
for those looking at this in the future, i cloned the Arc to pass to create_surface:

```rust
let surface: Surface = instance.create_surface(window_arc.clone()).unwrap();
```

i then grab the window handle with as_ref:

```rust
let size: PhysicalSize<u32> = window_arc.as_ref().inner_size();
```

**Reply by cwfitzgerald** [reply]
For the second part, you should be able to call methods directly on the Arc<Window> without needing as_ref()
